### PR TITLE
Sync obelix with workflowAPI

### PR DESCRIPTION
**Pull request by TomChv**
# Description

- Add workflowAPI into other dev environement
- Add workflowAPI seeder
- Add new TaskAction

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Old enumeration is not supported anymore

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [x] I have added sufficient documentation in the code

### CODE REVIEW

File: backend/protos-ts/task.proto [hunk 1]
```diff
@@ -20,10 +20,13 @@ enum TaskType {
...
-  GITHUB_ISSUE_CLOSE = 1; // Action & reaction
-  TIMER_DATE = 2;         // Action
-  TIMER_INTERVAL = 3;     // Action
+  GITHUB_PR_OPEN = 0;            // Action
+  GITHUB_ISSUE_OPEN = 1;         // Action
+  TIMER_DATE = 2;                // Action
+  TIMER_INTERVAL = 3;            // Action
+  GOOGLE_CALENDAR_NEW_EVENT = 4; // Reaction
+  GOOGLE_NEW_DOC = 5;            // Reaction
+  GOOGLE_NEW_SHEET = 6;          // Reaction
```
Comment (0xpanoramix) [inline backend/protos-ts/task.proto]: Please stay consistent in the naming of those fields.

File: backend/protos-ts/workflow/task.ts [hunk 2]
```diff
@@ -21,14 +21,20 @@ export enum TaskType {
...
   /** TIMER_DATE - Action */
   TIMER_DATE = 2,
   /** TIMER_INTERVAL - Action */
   TIMER_INTERVAL = 3,
+  /** GOOGLE_CALENDAR_NEW_EVENT - Reaction */
+  GOOGLE_CALENDAR_NEW_EVENT = 4,
+  /** GOOGLE_NEW_DOC - Reaction */
+  GOOGLE_NEW_DOC = 5,
+  /** GOOGLE_NEW_SHEET - Reaction */
+  GOOGLE_NEW_SHEET = 6,
```
Comment (0xpanoramix) [inline backend/protos-ts/workflow/task.ts:37]: Same as above.

File: backend/userAPI/src/task/dto/create-task.dto.ts [hunk 3]
```diff
@@ -6,10 +6,13 @@ export enum TaskType {
...
 export enum TaskAction {
-	GITHUB_PR_MERGE = 'GITHUB_PR_MERGE',
-	GITHUB_ISSUE_CLOSE = 'GITHUB_ISSUE_CLOSE',
+	GITHUB_PR_OPEN = 'GITHUB_PR_OPEN',
+	GITHUB_ISSUE_OPEN = 'GITHUB_ISSUE_OPEN',
 	TIMER_DATE = 'TIMER_DATE',
 	TIMER_INTERVAL = 'TIMER_INTERVAL',
+	GOOGLE_CALENDAR_NEW_EVENT = 'GOOGLE_CALENDAR_NEW_EVENT',
+	GOOGLE_NEW_DOC = 'GOOGLE_NEW_DOC',
+	GOOGLE_NEW_SHEET = 'GOOGLE_NEW_SHEET'
```
Comment (0xpanoramix) [inline backend/userAPI/src/task/dto/create-task.dto.ts:15]: Same as above.

File: backend/workflowAPI/prisma/migrations/20220219143000_add_enumeration/migration.sql [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/*
+  Warnings:
+
+  - The values [GITHUB_PR_MERGE,GITHUB_ISSUE_CLOSE] on the enum `TaskAction` will be removed. If these variants are still used in the database, this will fail.
+
+*/
+-- AlterEnum
+BEGIN;
+CREATE TYPE "TaskAction_new" AS ENUM ('GITHUB_PR_OPEN', 'GITHUB_ISSUE_OPEN', 'TIMER_DATE', 'TIMER_INTERVAL', 'GOOGLE_CALENDAR_NEW_EVENT', 'GOOGLE_NEW_DOC', 'GOOGLE_NEW_SHEET');
```
Comment (0xpanoramix) [inline backend/workflowAPI/prisma/migrations/20220219143000_add_enumeration/migration.sql]: Same as above.

File: backend/workflowAPI/prisma/schema.prisma [hunk 5]
```diff
@@ -16,10 +16,13 @@ enum TaskType {
...
-    GITHUB_ISSUE_CLOSE // Action & reaction
-    TIMER_DATE         // Action
-    TIMER_INTERVAL     // Action
+  GITHUB_PR_OPEN            // Action
+  GITHUB_ISSUE_OPEN         // Action
+  TIMER_DATE                // Action
+  TIMER_INTERVAL            // Action
+  GOOGLE_CALENDAR_NEW_EVENT // Reaction
+  GOOGLE_NEW_DOC            // Reaction
+  GOOGLE_NEW_SHEET          // Reaction
```
Comment (0xpanoramix) [inline backend/workflowAPI/prisma/schema.prisma]: Same as above.

File: protos/task.proto [hunk 6]
```diff
@@ -20,10 +20,13 @@ enum TaskType {
...
-  GITHUB_ISSUE_CLOSE = 1; // Action & reaction
-  TIMER_DATE = 2;         // Action
-  TIMER_INTERVAL = 3;     // Action
+  GITHUB_PR_OPEN = 0;            // Action
+  GITHUB_ISSUE_OPEN = 1;         // Action
+  TIMER_DATE = 2;                // Action
+  TIMER_INTERVAL = 3;            // Action
+  GOOGLE_CALENDAR_NEW_EVENT = 4; // Reaction
+  GOOGLE_NEW_DOC = 5;            // Reaction
+  GOOGLE_NEW_SHEET = 6;          // Reaction
```
Comment (0xpanoramix) [inline protos/task.proto]: Same as above.

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]
LGTM :rocket:

File: protos/task.proto (hunk 6 above)
Comment (TomChv) [reply protos/task.proto]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

File: backend/workflowAPI/prisma/schema.prisma (hunk 5 above)
Comment (TomChv) [reply backend/workflowAPI/prisma/schema.prisma]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

File: backend/workflowAPI/prisma/migrations/20220219143000_add_enumeration/migration.sql (hunk 4 above)
Comment (TomChv) [reply backend/workflowAPI/prisma/migrations/20220219143000_add_enumeration/migration.sql]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

File: backend/userAPI/src/task/dto/create-task.dto.ts (hunk 3 above)
Comment (TomChv) [reply backend/userAPI/src/task/dto/create-task.dto.ts:15]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

File: backend/protos-ts/workflow/task.ts (hunk 2 above)
Comment (TomChv) [reply backend/protos-ts/workflow/task.ts:37]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

File: backend/protos-ts/task.proto (hunk 1 above)
Comment (TomChv) [reply backend/protos-ts/task.proto]: Fixed in https://github.com/Cleopha/ifttt-like/pull/42/commits/5bd497b96dbec7651fb3c60fdd9d87ffbf3a89de

**Review by 0xpanoramix — APPROVED** [review]

**Pull request merged**